From 8d15e2350ac16f9a0b62fe61b79f8bf3fa157a74 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 22 Sep 2022 10:30:07 +0300
Subject: [PATCH 07/30] plat: s32g: Fix compilation warning

comparison of integer expressions of different signedness: 'int' and
'unsigned int'

Issue: ALB-9298
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_dt.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/plat/nxp/s32/s32_dt.c b/plat/nxp/s32/s32_dt.c
index b3e5dca6e..404c4c9e7 100644
--- a/plat/nxp/s32/s32_dt.c
+++ b/plat/nxp/s32/s32_dt.c
@@ -58,12 +58,18 @@ uint8_t fdt_get_status(int node)
 int dt_enable_clocks(void *fdt_addr, int node)
 {
 	const fdt32_t *cuint;
-	int len, i, index, ret;
+	int len = 0, ret;
 	uint32_t clk_id, clk_drv_id;
+	size_t ncells, i, index;
 	struct clk clk;
 
 	cuint = fdt_getprop(fdt_addr, node, "clocks", &len);
-	if (!cuint)
+	if (!cuint || len <= 0)
+		return 0;
+
+	ncells = (size_t)len / FDT_CLOCK_CELL_SIZE;
+	// Invalid data
+	if (!ncells)
 		return 0;
 
 	if (len % FDT_CLOCK_CELL_SIZE) {
@@ -72,8 +78,9 @@ int dt_enable_clocks(void *fdt_addr, int node)
 		return -EIO;
 	}
 
-	for (i = 0; i < len / FDT_CLOCK_CELL_SIZE; i++) {
+	for (i = 0; i < ncells; i++) {
 		index = i * 2;
+
 		clk_drv_id = fdt32_to_cpu(cuint[index]);
 		clk_id = fdt32_to_cpu(cuint[index + 1]);
 
-- 
2.17.1

